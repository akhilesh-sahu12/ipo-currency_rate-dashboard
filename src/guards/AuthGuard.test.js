import React from 'react';
import { render } from '@testing-library/react';
import AuthGuard from './AuthGuard';
import { AuthProvider } from '../context/AuthContext';
import { BrowserRouter as Router } from 'react-router-dom';

describe('AuthGuard component', () => {
  it('renders child routes when authenticated', () => {
    const { container } = render(
      <AuthProvider>
        <Router>
          <AuthGuard />
        </Router>
      </AuthProvider>
    );

    // Add assertions for your specific case
    // For example, check if child routes are rendered when authenticated
    expect(container).toMatchSnapshot();
  });

  it('redirects to /login when not authenticated', () => {
    // Mock your authentication state to be not authenticated
    // Render AuthGuard
    // Ensure that it redirects to /login
  });
});
