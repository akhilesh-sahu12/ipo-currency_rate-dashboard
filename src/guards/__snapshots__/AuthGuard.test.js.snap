// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthGuard component renders child routes when authenticated 1`] = `<div />`;
